src/OrchardCore.Modules/OrchardCore.Media/Indexing/MediaFieldIndexHandler.cs

```diff
@@ -39,9 +39,16 @@ public async override Task BuildIndexAsync(MediaField field, BuildFieldIndexCont
                 {
                     foreach (var key in context.Keys)
                     {
-                        foreach (var mediaText in field.MediaTexts)
+                        if (field.MediaTexts != null)
```

Comment:
Here, for backward compatibility, since this is a String Array it will return a NRE if nothing has been previously assigned to this field.